frontend/src/metabase-lib/v1/queries/NativeQuery.ts

```diff
@@ -158,10 +158,15 @@ export default class NativeQuery extends AtomicQuery {
   setDatabaseId(databaseId: DatabaseId): NativeQuery {
     if (databaseId !== this._databaseId()) {
       const metadataProvider = Lib.metadataProvider(databaseId, this._metadata);
+      const extras = Lib.nativeExtras(this._query());
       const newQuery = Lib.withDifferentDatabase(
         this._query(),
         databaseId,
         metadataProvider,
+        {
+          collection: null,
```

Comment:
Should we always be passing this? Or should we look at the query to see what's missing?

Reply:
what happens if you just pass `extras`?

Reply:
I mean, without spreading and without `collection: null`

Reply:
In my case extra's is `{}` so, so it's the same issue.